db/migrate/20201211202144_add_color_for_predicates.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class AddColorForPredicates < ActiveRecord::Migration[6.0]
+  def change
+    change_table :predicates do |t|
+      t.string :color
+    end
+
+    Predicate.all.each {|p| p.assign_color }
```

Comment:
Using business logic methods in migrations is generally a bad practice. If the `Predicate#assign_color` method is gone in the future, the migration will break. So, either a check is needed for it's existence or the assignment should be executed separately (via a Rake task or just manually via the Rails console).